core/src/main/java/org/elasticsearch/action/bulk/BulkProcessor.java

```diff
@@ -145,13 +137,13 @@ public Builder setFlushInterval(TimeValue flushInterval) {
         }
 
         /**
-         * Sets a custom backoff policy. The backoff policy defines how the bulk processor should handle retries of bulk requests internally
-         * in case they have failed due to resource constraints (i.e. a thread pool was full).
-         *
-         * The default is to back off exponentially.
-         *
-         * @see org.elasticsearch.action.bulk.BackoffPolicy#exponentialBackoff()
+         * Sets an optional thread pool on which to schedules flush actions and request retries
```

Comment:
should it not be required? If so I would move it to the constructor of the Builder and remove the setter for it.